File: challenges/medium/03_pagination/processor.py
def normalize(records):
    """
    Scale each record's 'value' to the range [0.0, 1.0]
    using the maximum value in the set as the divisor.
    Returns the normalized records.
    """
    if not records:
        return records
    max_val = max(r["value"] for r in records)
    if max_val == 0:
        return records
    for record in records:
        record["value"] = round(record["value"] / max_val, 2)
    return records

File: challenges/medium/09_pagination_js/processor.js
function normalize(records) {
  if (!records.length) return records;
  const maxVal = Math.max(...records.map(r => r.value));
  if (maxVal === 0) return records;
  for (const record of records) {
    record.value = Math.round((record.value / maxVal) * 100) / 100;   // bug: mutates in place
  }
  return records;
}

module.exports = { normalize };
